Add multi select pattern documentation (#1938)
hashicorp/design-system: website/docs/patterns/multi-select-patterns/partials/specifications/specifications.md

> ## Anatomy

Author: I think it still makes sense to treat this as a table because it's a nice way to present the information. But I've removed the "required/optional" statements and made this a bit more general/broad.

> ![Multi-select anatomy](/assets/patterns/multi-select-patterns/selection-anatomy.png)

Reviewer: In other parts the "Global selection options" is referred to as "Bulk selection", I think you have to update the image.

| Bulk selection | Optional; consolidates multiple selection functions into a single element. |

## Spacing

### Multi-select pattern

Within the Multi-select pattern itself, use a 16px horizontal gap between the message and the actions.

![Spacing within the multi-select pattern](/assets/patterns/multi-select-patterns/multi-select-pattern-spacing.png)

### In context

Within a larger context, use the same spacing recommendation of 16px vertical gap between the pattern, filter elements, and data set.

![Spacing in context with other elements](/assets/patterns/multi-select-patterns/multi-select-in-context-spacing.png)

## Basic layout

In a basic example of a Table, place the selection pattern above the top-right corner of the Table.

![A basic multi-select example in context](/assets/patterns/multi-select-patterns/multi-select-in-context-basic.png)

### Layout with a filter pattern

When composing a Multi-select pattern with a [Filter bar](/patterns/filter-patterns?tab=specifications#filter-bar-1) and [Applied filters](/patterns/filter-patterns?tab=specifications#applied-filters-1), maintain the sample placement.

> ![Multi-select with a Filter Pattern](/assets/patterns/multi-select-patterns/multi-select-in-context-filters.png)

Author: I intentionally scaled these up a bit; since the surrounding UI is meant to be more representative I wanted to make them stand out a bit more. I can reduce them if you feel they're a bit unrealistic

### In context

> Within a larger context, use a 16px vertical gap between the filter elements, multi-selection pattern, and table.

Reviewer 2: What's a "larger context"? Do you mean with multiple rows of information in the multi-select pattern?

I would mention vertical spacing vs. horizontal spacing which I think would help clarify what is meant.

Author: By larger context I mean when paired with other filtering patterns and functions around the table. I'll do a bit to clarify this

![Spacing in context with other elements](/assets/patterns/multi-select-patterns/multi-select-in-context-spacing.png)

## Layout

> In a basic example of a Table, place the selection pattern above the top-right corner of the Table.

Reviewer 2: ```suggestion
For a basic Table, place the selection pattern above the top-right corner of the Table.
```

Reviewer 2: Btw, what would a "basic" table be vs. a non-basic table for purposes of these examples?

Author: A basic table would be one that doesn't use filtering.

Reviewer 2: Maybe mention that to clarify?
